adapters_test.go

```diff
@@ -0,0 +1,240 @@
+	t.Run("returns empty slice when input txs is nil", func(t *testing.T) {
+		txs, err := monomer.AdaptCosmosTxsToEthTxs(nil)
+		require.NoError(t, err)
+		require.Empty(t, txs)
+	})
+
+	t.Run("returns empty slice when input txs is empty", func(t *testing.T) {
+		txs, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{})
+		require.NoError(t, err)
+		require.Empty(t, txs)
+	})
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, ethTxs, txBytes := generateTxsAndTxsBytes(t, test.depTxNum, test.nonDepTxNum)
+			txs, err := monomer.AdaptPayloadTxsToCosmosTxs(txBytes, test.signTx, test.from)
+			require.NoError(t, err)
+
+			adaptedTxs, err := monomer.AdaptCosmosTxsToEthTxs(txs)
+			require.NoError(t, err)
+			require.Equal(t, len(ethTxs), len(adaptedTxs))
+			for i := range ethTxs {
+				require.Equal(t, ethTxs[i].Hash(), adaptedTxs[i].Hash(), i)
+			}
+		})
+	}
+
+	t.Run("returns error when input contains invalid binary data", func(t *testing.T) {
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{[]byte("invalid")})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unexpected number of messages in Cosmos tx", func(t *testing.T) {
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, &codectypes.Any{}, nil)
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when L1 Attributes tx is not found", func(t *testing.T) {
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{})
+		require.NoError(t, err)
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, msgAny, nil)
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unable to unmarshal MsgL1Txs message", func(t *testing.T) {
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, &codectypes.Any{Value: []byte("invalid")}, nil)
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when MsgL1Tx contains only non-deposit txs", func(t *testing.T) {
+		txBytes, _, _ := generateTxsAndTxsBytes(t, 0, 5)
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{TxBytes: txBytes})
+		require.NoError(t, err)
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, msgAny, nil)
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("returns error when unable to unmarshal binary data within MsgL1Tx", func(t *testing.T) {
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{
+			TxBytes: [][]byte{[]byte("invalid")},
+		})
+		require.NoError(t, err)
+		depositSDKMsgBytes := generateDepositSDKMsgBytes(t, msgAny, nil)
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+}
+
+func signTxHelper(t *testing.T, sdkTx *tx.Tx, signTx monomer.TxSigner) *tx.Tx {
```

Comment:
You could probably just inline `signTxHelper` in `generateDepositSDKMsgBytes` since it's only used once